### PR TITLE
Lacrymology/xaiki/event emitter

### PR DESCRIPTION
**Pull request by Lacrymology**
Replayed relevant commits from #58, fixed conflicts, and made tests pass.

### CODE REVIEW

**Comment by xaiki** [comment]
is this for review ?

**Comment by Lacrymology** [comment]
There, no mixed functionality. createPlaylist always gets a callback, and it wouldn't even do that if it wasn't because it depends on an async function

File: Makefile [hunk 1]
```diff
@@ -58,8 +60,10 @@ test/videos/%.avi: test/images/%.png
 	avconv -loop 1 -f image2 -i $< -t 30 $@ &> /dev/null
 
 test/videos/%.mp4: test/images/%.png
-	avconv -loop 1 -f image2 -i $< -t 30 $@ &> /dev/null
+	${AVCONV} -loop 1 -f image2 -i $< -t 30 $@ &> /dev/null
 
 test: videos ${MOCHA} melted-restart
 	${NODE} ${MOCHA}
```
Comment (jmrunge) [inline Makefile:66]: Shouldnt we call "clean-test" after this line? Or is it ment to be runned alone?

File: drivers/playlists/mongo-driver.js [hunk 2]
```diff
@@ -38,78 +40,77 @@ function mongo_driver(conf) {
...
             }
         });
         channel.subscribe({backend: 'schedbackend', method: 'update'}, function(msg) {
             // I forward all create messages
-            self.createPlaylist(msg.model, drop_err(self.updatePlaylistCallback, console.log));
+            self.createPlaylist(msg.model, function(err, playlist) {
+                if( err )
+                    console.error("mongo-driver [ERROR]:", err);
+                else
+                    self.emit('create', playlist)
```
Comment (jmrunge) [inline drivers/playlists/mongo-driver.js]: Why are you emitting a 'create' event in the 'update' subscription?

**Comment by jmrunge** [comment]
You havent yet added to json driver the methods you added to mongo, dont you? (getPlaylists, setWindow ?)  Just for compatibility (not getting and undefined error when testing json).

File: Makefile (hunk 1 above)
Comment (xaiki) [inline Makefile:66]: there is no such target.

File: drivers/playlists/mongo-driver.js (hunk 2 above)
Comment (xaiki) [inline drivers/playlists/mongo-driver.js]: my mistake good catch

File: Makefile (hunk 1 above)
Comment (jmrunge) [inline Makefile:66]: Line 68, added by @Lacrymology :
+clean-test:
-  rm ${TEST_VIDEOS}

Comment (xaiki) [inline Makefile:66]: On 22 April 2013 01:27, Juan Martin Runge notifications@github.com wrote:

> In Makefile:
> 
> > @@ -58,8 +60,10 @@ test/videos/%.avi: test/images/%.png
> >     avconv -loop 1 -f image2 -i $< -t 30 $@ &> /dev/null
> > 
> >  test/videos/%.mp4: test/images/%.png
> > -   avconv -loop 1 -f image2 -i $< -t 30 $@ &> /dev/null
> > -   ${AVCONV} -loop 1 -f image2 -i $< -t 30 $@ &> /dev/null
> >   
> >   test: videos ${MOCHA} melted-restart
> >     ${NODE} ${MOCHA}
> 
> Line 68, added by @Lacrymology https://github.com/Lacrymology :
> +clean-test:
> - rm ${TEST_VIDEOS}
> 
> oh, cute.
> then no, it's pointless to regenerate the vids everytime.

## 

Niv Sardi

File: drivers/playlists/mongo-driver.js (hunk 2 above)
Comment (Lacrymology) [inline drivers/playlists/mongo-driver.js]: I.. don't know. By mistake? copy-paste?

File: Makefile (hunk 1 above)
Comment (jmrunge) [inline Makefile:66]: Ok! Agreed @Lacrymology ?

File: drivers/playlists/mongo-driver.js (hunk 2 above)
Comment (jmrunge) [inline drivers/playlists/mongo-driver.js]: Well... Can it be fixed? Or should we merge it this way and then fix it on another branch? (just kidding!)

File: Makefile (hunk 1 above)
Comment (Lacrymology) [inline Makefile:66]: I... didn't add that, I think it's @csrocha's

what I **do** think is we should call `make melted-restart` or `melted-kill` after tests

Comment (jmrunge) [inline Makefile:66]: I think is a good idea...

-----Original Message-----

File: drivers/playlists/mongo-driver.js (hunk 2 above)
Comment (xaiki) [inline drivers/playlists/mongo-driver.js]: fixed.

**Comment by xaiki** [comment]
rebased to apply on master.

**Comment by xaiki** [comment]
hum, actually, some of the changes are tricky for me to merge.
@Lacrymology can you rebase on master ?

**Comment by Lacrymology** [comment]
yes, no problem

**Comment by Lacrymology** [comment]
git rerere is doing NOTHING for me, just by the way

**Comment by Lacrymology** [comment]
rebased

**Comment by Lacrymology** [comment]
can we merge this?

**Comment by jmrunge** [comment]
@xaiki: Anything else to say? Will merge it tomorrow morning if no replies...

**Comment by xaiki** [comment]
for me it's ok.
merge it.

On 23 April 2013 00:34, Juan Martin Runge notifications@github.com wrote:

> @xaiki https://github.com/xaiki: Anything else to say? Will merge it
> tomorrow morning if no replies...
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/inaes-tic/mbc-mosto/pull/83#issuecomment-16837491
> .

## 

Niv Sardi

**Pull request merged**